PFR_solver_CANTERA.m

```diff
@@ -0,0 +1,43 @@
+function [F] = PFR_solver_CANTERA(x,initial,gas,mdot,A_in,dAdx,k)
+
+rho = initial(1);
+T = initial(2);
+Y = initial(3:length(initial));
+
+if k==1
+    A = A_in-k*dAdx*x;
+elseif k==-1|k==0
+    A=A_in+k*dAdx*x;
+    dAdx=-dAdx;
+end
+
+MW_mix = meanMolecularWeight(gas);
+Ru = 8314;
+R = Ru/MW_mix;
+nsp = nSpecies(gas);
+vx = mdot/(rho*A);
+P = rho*R*T;
+if k~=0
+    set(gas,'Temperature',T,'Density',rho,'MassFractions',Y); % the gas is set to the corresponding properties during each iteration of the ode loop
+else
+    set(gas,'Temperature',T,'Pressure',P,'MassFractions',Y); % the gas is set to the corresponding properties during each iteration of the ode loop

```

Comment:
Why use pressure to set the state when `k = 0`? Density should be equally as valid, right?

Reply:
Yes. I changed it